cmd/primateRecon/primateRecon.go

```diff
@@ -5,29 +5,133 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/expandedTree"
 	"github.com/vertgenlab/gonomics/fasta"
 	"github.com/vertgenlab/gonomics/reconstruct"
 	"log"
 )
 
-func quickPrimateRecon(infile string, outfile string, messyToN bool) {
+// likelihoodsToBase takes the un-normalized likelihoods for A, C, G, T as well
+// as the index of the human base (0 for A, 1 for C, etc), and the probability
+// threshold for when we will call it the mle base instead of the human base
+// and gives back the reconstructed base for the hca
+func likelihoodsToBase(likes []float64, humanBase dna.Base, probThreshold float64, nonHumanProbThreshold float64, messyToN bool) dna.Base {
+	var total, nonHumanTotal, bestProb float64
+	var i int
+	var answer dna.Base
+
+	if messyToN || humanBase == dna.Gap {
+		answer = dna.N
+	} else {
+		answer = humanBase
+	}
+
+	for i = range likes {
+		total += likes[i]
+		if i != int(humanBase) {
+			nonHumanTotal += likes[i]
+		}
+	}
+
+	for i = range likes {
+		if likes[i]/total >= probThreshold && likes[i] > bestProb && nonHumanTotal/total >= nonHumanProbThreshold {
+			bestProb = likes[i]
+			answer = dna.Base(i)
+		}
+	}
+	return answer
+}
+
+func baseIsPresent(b dna.Base) bool {
+	if dna.DefineBase(b) || b == dna.N {
+		return true
+	}
+	return false
+}
+
+func hcaIsPresent(human, bonobo, chimp, gorilla, organutan dna.Base) bool {
+	if baseIsPresent(human) && (baseIsPresent(bonobo) || baseIsPresent(chimp)) {
+		return true
+	}
+	if (baseIsPresent(human) || baseIsPresent(bonobo) || baseIsPresent(chimp)) && (baseIsPresent(gorilla) || baseIsPresent(organutan)) {
```

Comment:
in the manual algorithm, we would return N if human and gorilla were present but not bonobo and chimp, with the worry of estimating over the longer branch. Does the usage of branch lengths in the ML help us here? Right now it looks like we'll evaluate a base at a position with no bonobo or chimp